app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -544,6 +544,18 @@ public void updateFailingUrl(String url, boolean updateFromError) {
                     this.failingUrl = url;
                 }
             }
+
+            @Override
+            public void onReceivedTitle(WebView view, String title) {
+                if(!mostOldCallbacksHaveFinished) {
+                    return;
+                }
+
+                if (!UrlUtils.isInternalErrorURL(view.getUrl())
```

Comment:
Do you feel like it's not a good idea to move if (!UrlUtils.isInternalErrorURL(view.getUrl())) into 
```
updateURL(final String url) {
    if(UrlUtils.isInternalErrorURL(view.getUrl(url))) {
        return;
    }
}
```